src/core/builders/vite/index.ts

```diff
@@ -157,14 +141,14 @@ export async function createViteBuilder(
       build: {
         rollupOptions: {
           input: entrypoints.reduce<Record<string, string>>((input, entry) => {
-            input[entry.name] = entry.inputPath;
+            input[entry.name] = getEntrypointEntry(entry);
             return input;
           }, {}),
           output: {
             // Include a hash to prevent conflicts
             chunkFileNames: 'chunks/[name]-[hash].js',
-            // Include a hash to prevent conflicts
-            entryFileNames: 'chunks/[name]-[hash].js',
+            // Place JS entrypoints in main directory without a hash. (popup.html & popup.js are next to each other)
+            entryFileNames: '[name].js',
```

Comment:
Breaking change to the build output format.

Reply:
This change ensures the ESM background entrypoint (a JS file, not HTML), is output to the root of the build directory as well, just like the non-ESM background.

Reply:
@aklinker1 Is it possible to control the output path using wxt.config.ts?

To have one build for dev and prod.
So when I build for dev it doesn't overwrite previously built one for prod.
```
.output/
  dev-chrome-mv3/
    chunks/
      popup-<hash>.js
    popup.html
```
And also

```
.output/
  prod-chrome-mv3/
    chunks/
      popup-<hash>.js
    popup.html
```

Reply:
@lionelhorn In a hacky way, yes. You could set an environment variable, and reference it in the `wxt.config.ts` file while setting the `outDir`. WXT doesn't provide any options for changing the target directories inside the output directory.

```sh
DEV=true wxt
wxt build
```

```ts
import { defineConfig } from 'wxt';

export default defineConfig({
  outDir: process.env.DEV === "true" ? ".output/dev" : ".output"
});
```

This would give you something like:

```
.output/
   chrome-mv3/
   dev/
     chrome-mv3/
     firefox-mv2/
   firefox-mv2/
```